service/history/api/startworkflow/api.go

```diff
@@ -506,17 +507,44 @@ func (s *Starter) respondToRetriedRequest(
 	return s.generateResponse(runID, mutableStateInfo.workflowTask, events)
 }
 
+func (s *Starter) getWorkflowStartTime(ctx context.Context, runID string) (start time.Time, retError error) {
+	mutableState, releaseFn, retError := s.getMutableState(ctx, runID)
+	var workflowStartTime time.Time
+
+	defer func() {
+		// do we need "recover" here?
+		releaseFn(retError)
+	}()
+
+	if retError != nil {
+		return workflowStartTime, retError
+	}
+	if mutableState == nil {
+		return workflowStartTime, serviceerror.NewInternal("Can't load mutable state")
+	}
+
+	workflowStartTime = mutableState.GetExecutionInfo().StartTime.AsTime()
+	return workflowStartTime, nil
+}
+
 // getMutableStateInfo gets the relevant mutable state information while getting the state for the given run from the
 // workflow cache and managing the cache lease.
-func (s *Starter) getMutableStateInfo(ctx context.Context, runID string) (*mutableStateInfo, error) {
-	mutableState, err := s.getMutableState(ctx, runID)
-	if err != nil {
-		return nil, err
+func (s *Starter) getMutableStateInfo(ctx context.Context, runID string) (mutableSateInfo *mutableStateInfo, retError error) {
+	mutableState, releaseFn, retError := s.getMutableState(ctx, runID)
+
+	defer func() {
+		// do we need "recover" here?
```

Comment:
Good question. I would say it's fine to keep it as it was before and handle it when we understand whether it happens and, hopefully, why. If you do want to recover, maybe you could recover, release, and then re-panic if the recover result was not nil.

Reply:
I don't think we need recover here.